### PR TITLE
[SAI-PTF] Add ecmp test

### PR DESCRIPTION
**Pull request by zhiyuan0112**
## Description of PR
1. Config route and neighbor for nexthop group.
2. Add ecmp hash fileds ceses.
3. Modify Ecmp to NexthopGroup.

## Test
Tested in DUT with the pipeline.

Signed-off-by: zhiyuan-ms [t-zhliang@microsoft.com](mailto:t-zhliang@microsoft.com)

### CODE REVIEW

File: test/sai_test/config/route_configer.py [hunk 1]
```diff
@@ -314,6 +409,47 @@ def create_neighbor_by_rif(self, nexthop_device: Device, rif, no_host=True):
             self.test_obj.dut.neighborv6_list.append(nbr_entry_v6)
         return nbr_entry_v4, nbr_entry_v6
 
+    def create_nexthop_group_by_nexthops(self, nexthopv4_list: List[Nexthop], nexthopv6_list: List[Nexthop], lag_list: List[Lag], dest_device: Device):
```
Comment (richardyu-ms) [inline test/sai_test/config/route_configer.py:412]: Need more doc. Attrs returns

File: test/sai_test/config/route_configer.py [hunk 2]
```diff
@@ -118,6 +122,57 @@ def t0_route_config_helper(
         #set expected dest T1
         test_obj.t1_list[2][100].l3_lag_obj = test_obj.dut.lag_list[1]
 
+    if is_create_route_for_nhopgrp:
```
Comment (richardyu-ms) [inline test/sai_test/config/route_configer.py:125]: Print some message as log

File: test/sai_test/config/route_configer.py (hunk 1 above)
Comment (richardyu-ms) [inline test/sai_test/config/route_configer.py:412]: This method will create nexthop base on lag, please specified it from method name

Comment (zhiyuan0112) [reply test/sai_test/config/route_configer.py:412]: No, this method creates nexthop group based on nexthops, (i.e., `nexthopv4_list` and `nexthopv6_list`). Passing `lag_list` is to assign the values to `nexthop_groupv4` and `nexthop_groupv6`, which are new added in the `Lag` class.

**Comment by zhiyuan0112** [comment]
## Description
1. Modify code following comments
2. Add IngressNoDiffTestV4 and RemoveLagEcmpTestV4
3. Fix a bug in `create_nexthop_group_by_nexthops` method. The `nhp_grp_members` were incorrect in previous code. 

Signed-off-by: zhiyuan-ms [t-zhliang@microsoft.com](mailto:t-zhliang@microsoft.com)

File: test/sai_test/config/route_configer.py (hunk 1 above)
Comment (richardyu-ms) [reply test/sai_test/config/route_configer.py]: assign it outside this method? then this create nexthop group just will be more specific

File: test/sai_test/sai_ecmp_test.py [hunk 3]
```diff
@@ -0,0 +1,893 @@
...
+                                         ip_ttl=63)
+
+            send_packet(self, self.dut.port_obj_list[1].dev_port_index, pkt)
+            verify_any_packet_any_port(self, [exp_pkt1, exp_pkt2, exp_pkt3], recv_dev_port_idxs)
+
+    def runTest(self):
+        self.test_lag_ecmp_remove()
+
+    def tearDown(self):
+        super().tearDown()
```
Comment (richardyu-ms) [inline test/sai_test/sai_ecmp_test.py:893]: need to add back the nhop member to restore the environment.

File: test/sai_test/sai_ecmp_test.py [hunk 4]
```diff
@@ -0,0 +1,893 @@
...
+    def test_lag_ecmp_remove(self):
+        """
+        1. Remove the next hop from next-hop group in test_ecmp: next-hop with IP ``DIP:10.1.3.100`` on LAG3 
+        2. Generate Packets, with different source IPs as ``SIP:192.168.0.1-192.168.0.10`` 
+        3. Change other elements in the packets, including ``DIP:192.168.60.1`` ``L4_port``
+        4. Verify Packets only can be received on LAG1 and LAG2, with ``SMAC: SWITCH_MAC_2`` (check loadbalanced in LAG and ECMP)
+        """
+        print("Ecmp remove lag test")
+        
+        self.route_configer.remove_nhop_member_by_lag_idx(
```
Comment (richardyu-ms) [inline test/sai_test/sai_ecmp_test.py:850]: move those configuration steps into setup method.

**Pull request closed without merging**